query-engine/connector-test-kit-rs/query-test-macros/src/args/utils.rs

```diff
@@ -27,6 +13,14 @@ pub fn validate_suite(suite: &Option<String>, on_module: bool) -> Result<(), dar
 
 pub fn connectors_to_test(only: &OnlyConnectorTags, exclude: &ExcludeConnectorTags) -> Vec<ConnectorTag> {
     if !only.is_empty() {
+        if !exclude.is_empty() {
```

Comment:
Because of the comment above, this can't work and is the reason why we didn't allow only + exclude before. You're filtering on connector tags here, not versions. You'd need to deep-diff the connector tags + versions, not only the connector tags, as Cockroach is not a connector.